File: samples/jafu-reactive-data-r2dbc/src/test/java/com/sample/IntegrationTests.java
package com.sample;

import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.http.MediaType;
import org.springframework.test.web.reactive.server.WebTestClient;

public class IntegrationTests {

	private WebTestClient client = WebTestClient.bindToServer().baseUrl("http://localhost:8181").build();

	private ConfigurableApplicationContext context;

	@BeforeAll
	void beforeAll() {
		context = Application.app.run("test");
	}

	@Test
	void requestHttpApiEndpoint() {
		client.get().uri("/").exchange()
				.expectStatus().is2xxSuccessful()
				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE);
	}

	@Test
	public void requestHttpApiEndpointForListingAllUsers() {
		client.get().uri("/api/user").exchange()
				.expectStatus().is2xxSuccessful()
				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE)
				.expectBodyList(User.class)
				.hasSize(3);
	}

	@Test
	public void requestHttpApiEndpointForGettingOneSpecificUser() {
		client.get().uri("/api/user/bclozel").exchange()
				.expectStatus().is2xxSuccessful()
				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE)
				.expectBody(User.class)
				.isEqualTo(new User("bclozel", "Brian", "Clozel"));
	}

	@AfterAll
	void afterAll() {
		context.stop();
	}
}

File: samples/kofu-reactive-redis/src/test/kotlin/com/sample/IntegrationTests.kt
package com.sample

import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.http.MediaType
import org.springframework.test.web.reactive.server.WebTestClient
import org.springframework.test.web.reactive.server.expectBody
import org.springframework.test.web.reactive.server.expectBodyList
import org.testcontainers.containers.GenericContainer

class IntegrationTests {

	private val client = WebTestClient.bindToServer().baseUrl("http://localhost:8181").build()

	private lateinit var context: ConfigurableApplicationContext

	private val redisContainer = object : GenericContainer<Nothing>("redis:5") {}

	@BeforeAll
	fun beforeAll() {
		redisContainer.withExposedPorts(6379)
		redisContainer.start()
		val properties = ApplicationProperties(
				redisPort = redisContainer.firstMappedPort,
				redisHost = redisContainer.containerIpAddress,
				serverPort = 8181
		)
		context = app(properties).run()
	}

	@Test
	fun `Request root endpoint`() {
		client.get().uri("/").exchange()
				.expectStatus().is2xxSuccessful
				.expectHeader().contentType("text/html;charset=UTF-8")
	}

	@Test
	fun `Request HTTP API endpoint`() {
		client.get().uri("/api/user").exchange()
				.expectStatus().is2xxSuccessful
				.expectHeader().contentType(MediaType.APPLICATION_JSON_VALUE)
				.expectBodyList<User>()
				.hasSize(3)
	}

	@AfterAll
	fun afterAll() {
		context.close()
		redisContainer.stop()
	}
}